lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -1344,6 +1337,38 @@ migrateWallet srcCtx sheCtx (ApiT srcWid) (ApiT sheWid) migrateData = do
             #pendingSince
   where
     pwd = coerce $ getApiT $ migrateData ^. #passphrase
+    addrs = getApiT . fst <$> migrateData ^. #addresses
+
+-- | Transform the given set of migration coin selections (for a source wallet)
+--   into a set of coin selections that will migrate funds to the specified
+--   target addresses.
+--
+-- Each change entry in the specified set of coin selections is replaced with a
+-- corresponding output entry in the returned set, where the output entry has a
+-- address from specified addresses.
+--
+-- The difference between the size of coin selection and the number of
+-- specified addresses is solved by cycling specified addresses.
```

Comment:
```suggestion
-- If the number of outputs in the specified coin selection is greater than
-- the number of addresses in the specified address list, addresses will be
-- recycled in order of their appearance in the original list.
```

Reply:
changed in https://github.com/input-output-hk/cardano-wallet/pull/1664/commits/cf2b9e58c71fd7e80f8a9e1b9073348dfba3a67e